src/compiler/binder.ts

```diff
@@ -943,13 +967,36 @@ namespace ts {
                 expression.kind === SyntaxKind.FalseKeyword && flags & FlowFlags.TrueCondition) {
                 return unreachableFlow;
             }
+
+            if (flags & FlowFlags.TrueCondition) {
+                // If the antecedent is an optional chain, only the Present branch can exist on the `true` condition
+                const presentFlow = findAntecedent(currentFlow, FlowFlags.Present);

```

Comment:
`findAntecedent` (vs operating on all antecedents in some way) seems _slightly_ wrong. There are cases, I think, where a node could have multiple `Present` flagged antecedents, like at the start of a `finally` block, eg
```ts
try {
    if (a?.b.c) {
        // ...
    }
    if (a.b?.c) {
        // ...
    }
}
finally {
    if (a?.b?.c) {} // every flow node in the `try` is an antecedent of the start of the `finally`.
}
```
and it's unclear what should be done in those cases (or what the visible effects would be of doing something different than picking the first). In any case, I think filtering the label to all `Present` nodes, rather than `find`ing the first is _probably_ more correct.

Reply:
Don't those all become antecedents of a `PreFinally` node? They wouldn't hit this case since it's only triggered in `createFlowCondition` and `createFlowOptionalChain`. That said, I will switch it to only *excluding* `Missing` nodes rather than only *including* `Present` ones.